homeassistant/components/lastfm/config_flow.py

```diff
@@ -0,0 +1,222 @@
+"""Config flow for LastFm."""
+from __future__ import annotations
+
+from typing import Any
+
+from pylast import LastFMNetwork, User, WSError
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    OptionsFlowWithConfigEntry,
+)
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+)
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_MAIN_USER, CONF_USERS, DOMAIN
+
+PLACEHOLDERS = {"api_account_url": "https://www.last.fm/api/account/create"}
+
+CONFIG_SCHEMA: vol.Schema = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): str,
+        vol.Required(CONF_MAIN_USER): str,
+    }
+)
+
+
+def get_lastfm_user(api_key: str, username: str) -> User:
+    """Get lastFM User."""
+    return LastFMNetwork(api_key=api_key).get_user(username)
+
+
+def validate_lastfm_user(user: User) -> dict[str, str]:
+    """Return error if the user is not correct. None if it is correct."""
+    errors = {}
+    try:
+        user.get_playcount()
+    except WSError as error:
+        if error.details == "User not found":
+            errors["base"] = "invalid_account"
+        elif (
+            error.details
+            == "Invalid API key - You must be granted a valid key by last.fm"
+        ):
+            errors["base"] = "invalid_auth"
+        else:
+            errors["base"] = "unknown"
+    except Exception:  # pylint:disable=broad-except
+        errors["base"] = "unknown"
+    return errors
+
+
+class LastFmConfigFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Config flow handler for LastFm."""
+
+    data: dict[str, Any] = {}
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> LastFmOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return LastFmOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Initialize user input."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self.data = user_input.copy()
+            main_user = get_lastfm_user(
```

Comment:
I think this needs to be inside a try block as it can probably raise?

Reply:
As you have both `get_lastfm_user` and `validate_lastfm_user` function it's probably cleaner if you have the try-block here (also below where you have the functions) instead of in the respective function

Reply:
You mean moving the try catch from the validate method to the place where we validate? That could be done of course, but since we also do this in the Options flow, it's a bit iffy, I like to don't repeat code haha

Reply:
For the raise for get_user, what I've seen is that it just creates the User object. Like it doesn't do requests when you make the object itself, so that's why we're validating with get playcount

Reply:
In that case it seems as you can ignore.
I would have assumed the library would have done something with `get_lastfm_user` but if that's not the case it seems logical to keep it as is